metrics-report-generator/src/main/scala/it/pagopa/interop/metricsreportgenerator/Main.scala

```diff
@@ -25,34 +24,64 @@ object Main extends App {
 
   logger.info("Starting metrics report generator job")
 
-  val blockingThreadPool: ExecutorService = Executors.newFixedThreadPool(Runtime.getRuntime.availableProcessors())
-
-  implicit val blockingEC: ExecutionContextExecutor     = ExecutionContext.fromExecutor(blockingThreadPool)
-  implicit val fileManager: FileManager                 = FileManager.get(FileManager.S3)(blockingEC)
-  implicit val readModelService: ReadModelService       = new MongoDbReadModelService(
-    ApplicationConfiguration.readModelConfig
-  )
-  implicit val dateTimeSupplier: OffsetDateTimeSupplier = OffsetDateTimeSupplier
-
-  val fileUtils: FileUtils     = new FileUtils(fileManager, dateTimeSupplier)
-  val fileWriters: FileWriters = new FileWriters(fileUtils, dateTimeSupplier)
-
-  def execution(): Future[Unit] = for {
-    activeAgreements <- Utils.retrieveAllActiveAgreements(ReadModelQueries.getActiveAgreements, 0, Seq.empty)
-    purposes         <- Utils.retrieveAllPurposes(ReadModelQueries.getPurposes, 0, Seq.empty)
-    agreementRecords = AgreementRecord.join(activeAgreements, purposes)
-    _ <- fileWriters.agreementsJsonWriter(agreementRecords)
-    _ <- fileWriters.agreementsCsvWriter(agreementRecords)
-  } yield ()
-
-  def run(): Future[Unit] = execution()
-    .recover(ex => logger.error("There was an error while running the job", ex))(global)
-    .andThen { _ =>
-      fileManager.close()
-      readModelService.close()
-      blockingThreadPool.shutdown()
-    }(global)
-
-  Await.result(run(), Duration.Inf): Unit
+  def resources(implicit
+    ec: ExecutionContext
+  ): Future[(ExecutorService, ExecutionContextExecutor, FileManager, ReadModelService, Jobs, Configuration)] = for {
+    config <- Configuration.read()
+    es     <- Future(Executors.newFixedThreadPool(1.max(Runtime.getRuntime.availableProcessors() - 1)))
+    blockingEC = ExecutionContext.fromExecutor(es)
+    fm   <- Future(FileManager.get(FileManager.S3)(blockingEC))
```

Comment:
`.get` isn't already returning `Future`?

Reply:
Nope, it's not